### PR TITLE
chore(ONYX-1233): cleanup the code for My Collection "List" view option

### PR DESCRIPTION
**Pull request by dariakoko**
This PR resolves [ONYX-1233] 

### Description

Cleanup My Collection "List" view options as this feature is no longer used

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- Cleanup My Collection "List" view options as this feature is no longer used -daria

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[ONYX-1233]: https://artsyproduct.atlassian.net/browse/ONYX-1233?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Dev changes (Cleanup My Collection "List" view options as this feature is no longer used -daria)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against f8960f56161953bbcace9eea0c89fb13c42aee0f
</p>

File: src/app/Components/ArtworkFilter/ArtworkFilterOptionsScreen.tsx [hunk 1]
```diff
@@ -99,7 +98,7 @@ export const ArtworkFilterOptionsScreen: React.FC<
     localFilterOptions
   ).concat(aggregateFilterOptions)
 
-  const sortedFilterOptions = filterOptions
+  const sortedFilterOptions = filterOptions // here
```
Comment (olerichter00) [inline src/app/Components/ArtworkFilter/ArtworkFilterOptionsScreen.tsx:101]: 👀 

```suggestion
  const sortedFilterOptions = filterOptions
```

**Review by olerichter00 — APPROVED** [review]
🙏

File: src/app/Scenes/MyCollection/Components/MyCollectionCollectedArtistsView.tests.tsx [hunk 2]
```diff
@@ -1,93 +1,40 @@
...
-  let mockEnvironment: ReturnType<typeof createMockEnvironment>
-
-  beforeEach(() => {
-    mockEnvironment = createMockEnvironment()
-  })
-
-  const TestRenderer = () => {
-    return (
+  const { renderWithRelay } = setupTestWrapper<MyCollectionCollectedArtistsViewTestsQuery>({
+    Component: (props: any) => (
```
Comment (MounirDhahri) [inline src/app/Scenes/MyCollection/Components/MyCollectionCollectedArtistsView.tests.tsx:11]: 👏

File: src/app/Scenes/MyCollection/MyCollection.tsx [hunk 3]
```diff
@@ -369,8 +367,6 @@ export const MyCollectionQueryRenderer: React.FC = () => {
 }
 
 export const MyCollectionPlaceholder: React.FC = () => {
-  const viewOption = GlobalStore.useAppState((state) => state.userPrefs.artworkViewOption)
```
Comment (MounirDhahri) [inline src/app/Scenes/MyCollection/MyCollection.tsx:372]: This seems to be no longer used, if so let's write a migration to remove it from the `userPrefs` model

**Review by MounirDhahri — APPROVED** [review]
LGTM - only one small request

File: src/app/Scenes/MyCollection/Components/MyCollectionCollectedArtistsView.tests.tsx (hunk 2 above)
Comment (araujobarret) [reply src/app/Scenes/MyCollection/Components/MyCollectionCollectedArtistsView.tests.tsx:11]: nit: you can just `({ me }) => (` here
if TS complains you can add to your test query `me @required(action: NONE) {` making sure `me` won't be null 😉

**Review by araujobarret — APPROVED** [review]

**Review by olerichter00 — APPROVED** [review]

File: src/app/Scenes/MyCollection/MyCollection.tsx (hunk 3 above)
Comment (dariakoko) [reply src/app/Scenes/MyCollection/MyCollection.tsx:372]: That's true, just added the migration and tests for two values at once - `artworkViewOption` and `artistViewOption`

**Review by MounirDhahri — APPROVED** [review]
Thanks for making the changes - LGTM
